### PR TITLE
Fix errors

### PR DESCRIPTION
**Pull request by MazeXP**
This PR allows VS and Rider to properly load projects using the Remora.Sdk without any build errors.

## Changes
 - Use msbuildproj for Sdk projects as per convention.
 - Changed ProjectGUIDs for test projects to reflect .NET projects (Solution had NetFx GUIDs)
 - Write stylecop.json file using WriteLinesToFile (handles file locking scenario better)
 - Moved import of base Sdk targets to the end to allow overriding of used properties

## Fixes
 - https://github.com/Remora/Remora.Discord/issues/249
 - Image
 - Image
 - Image

### CODE REVIEW

File: Tests/Directory.Build.targets [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<Project>
```
Comment (Nihlus) [inline Tests/Directory.Build.targets:1]: Is this file required?

Comment (MazeXP) [reply Tests/Directory.Build.targets:1]: It is required to avoid leaking the roots `Directory.Build.targets` into the tests project.

File: Tests/Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<Project>
+    <PropertyGroup>
+        <LegalAuthor>Jarl Gullberg</LegalAuthor>
+        <LegalEmail>jarl.gullberg@gmail.com</LegalEmail>
+        <ExecutableFrameworks>net6.0;net7.0</ExecutableFrameworks>
```
Comment (Nihlus) [inline Tests/Directory.Build.props:5]: Remove `net7.0` here - the runner uses .NET 6.0 at the moment.
